pages/guides/frontend-migrations/pf-4-to-pf5.mdx

```diff
@@ -0,0 +1,494 @@
+
+Now that dependencies are up to date and build is configured, it is time to migrate the source code. Follow the [official patternfly migration guide](https://www.patternfly.org/get-started/upgrade) to migrate your patternfly modules. 
+
+It is highly recommended to **run the codemods first** for both the JS and CSS. it will save you a lot of time.
+
+## FEC Migration guide
+
+There are only a few breaking changes for frontend components. The majority of them enforce more strict props. Almost all components remained unchanged.
+
+Please follow [this guide](/guides/frontend-migrations/frontend-components-3-to-5) to learn about the changes.
+
+## Verifying updates
+
+Now that your source files are migrated, you should be able to start the development server the same way you are used to.
+
+If you see no build errors, you are done with the source code migration.
+
+### The build outputs an about about missing source files!
+
+There is a possibility that some PF source files can't be resolved. The error can look similar to this:
+
+```sh
+ERROR in ./src/components/SourceDetail/ApplicationKebab.js
+Module build failed (from ./node_modules/ts-loader/index.js):
+Error: Unable to find source file for module KebabToggle! The module likely does not have unique file as is included within another file. Please add the entry into the COMPONENTS_CACHE in FEC repository
+    at findComponentModule (/home/username/path/to/repository/node_modules/@redhat-cloud-services/tsc-transform-imports/index.js:115:15)
+    at /home/username/path/to/repository/node_modules/@redhat-cloud-services/tsc-transform-imports/index.js:155:33
+    at Array.reduce (<anonymous>)
+    at createDynamicReactCoreImports (/home/username/path/to/repository/node_modules/@redhat-cloud-services/tsc-transform-imports/index.js:152:30)
+    at visitor (/home/username/path/to/repository/node_modules/@redhat-cloud-services/tsc-transform-imports/index.js:203:24)
+    at visitArrayWorker (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:86542:51)
+    at visitNodes2 (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:86513:21)
+    at visitLexicalEnvironment (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:86569:18)
+    at visitEachChildOfSourceFile (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:87758:13)
+    at Object.visitEachChild (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:86730:35)
+ @ ./src/components/SourceDetail/ApplicationsCard.js 20:0-50 180:71-87
+ @ ./src/pages/Detail.js 6:0-75 34:36-52
+ @ ./src/Routing.js 8:32-95
+ @ ./src/App.js 7:0-32 33:44-51
+ @ ./src/AppEntry.js 5:0-24 12:24-27
+ @ container entry ./RootApp[0]
+
+```
+
+The patched build checks each patternfly module and tries to resolve their source files. If it fails to find them, it will throw the above error. (The resolution is cached so it will not check one file more than once)
```

Comment:
```suggestion
The patched build checks each PatternFly module and tries to resolve their source files. If it fails to find them, it will throw the above error. The resolution is cached so it will not check one file more than once.
```